### DISCUSSION TITLE
Themed archive page worked, then it didn't

### DISCUSSION THREAD

**Discussion by seanvandermolen** [impress-org/givewp#7320] [Developer Q&A] [answered]
1.  I original used the standard archive/page query which worked, then switched to the example on the website because I'm going to want to exclude a post by id.  This worked.
 ` 
<?php if (have_posts()) : ?>
<?php while (have_posts()) : the_post(); ?>
`

2.  I then went to my test donation forms and there was a prompt asking me to upgrade two of them, which I did, then I added a goal because I wanted to customize the goal code.  I also added the section // Output the goal to my archive template.  At this point the archive stopped working an it went back to using the template/ iframe.

3.  What is the purpose of using an iframe?

![image](https://github.com/impress-org/givewp/assets/7504775/a30803ce-eece-4077-b003-f4cb938700c7)

archive-give_forms.php

`                        <?php
                        /**
                         *  Display only Give donation forms
                         */
                        $args = array(
                            'post_type'      => 'give_forms',
                            'posts_per_page' => 10,
                        );

                        $wp_query = new WP_Query( $args );

                        ?>
                        <?php if ( $wp_query->have_posts() ) : ?>
                            <?php while ( $wp_query->have_posts() ) : $wp_query->the_post(); ?>
                                <div class="col col-lg-4 col-md-12 col-12">
                                    <div class="causes-item">
                                        <?php if (has_post_thumbnail()) : ?>
                                            <div class="image">
                                                <a href="<?php the_permalink(); ?>">
                                                
                                                <?php the_post_thumbnail('medium', array('class' => 'card-img-lg-start')); ?>
                                                
                                                </a>
                                            </div>
                                        <?php endif; ?>
                                        <div class="content">
                                            <h2><a class="" href="<?php the_permalink(); ?>"><?php the_title(); ?></a></h2>
                                            <?php the_excerpt() ?>
                                            <?php 
                                                $id = get_the_ID();
                                                // Output the goal
                                                $goal_option = get_post_meta( $id, '_give_goal_option', true );
                                                if ( $goal_option == 'yes' ) {
                                                    $shortcode = '[give_goal id="' . $id . '"]';
                                                    echo do_shortcode( $shortcode );
                                                }
                                            ?>
                                            <h3 class="my-give-archive-donate-now-link"><a href="<?php echo get_the_permalink(); ?>">Donate Now</a></h3>
                                        </div>
                                    </div>
                                </div>
                            <?php endwhile; ?>
                        <?php endif; ?>`

give.php

`<div id="content" class="site-content <?= bootscore_container_class(); ?> p-0">
    <div id="primary" class="content-area">
        <?php the_post(); ?>
        <section class="wpo-page-title">
            <div class="container">
                <div class="row">
                    <div class="col col-xs-12">
                        <div class="wpo-breadcumb-wrap">
                        
                            <h2><?php the_title(); ?></h2>

                        </div>
                    </div>
                </div> 
            </div> 
        </section>

        
        <?php bs_after_primary(); ?>
        
        <section class="wpo-blog-single-section wpo-blog-single-left-sidebar-section section-padding">
            <div class="container">
                <div class="row">
                    <div class="col col-lg-8 col-12 order-lg-2">
                        <div class="wpo-blog-content">
                        <?php give_get_template_part( 'templates', 'single-give-form' ); ?>
                        </div>
                    </div>
                    <?php get_sidebar('child'); ?>
                </div>
            </div> 
        </section>
        
    </div>
</div>`

**Comment by seanvandermolen** [comment]
Well, I deleted the forms and recreated them without using the new form builder and it worked. 

Is that the expected behavior?

**Comment by kjohnson** [comment, marked as answer]
The iframe is used to reduce conflicts with themes and other plugins. The general approach that we take is to embed Give donation forms into content.

As for adding donation forms to a page, we have included custom blocks which service this purpose, such as the Form Grid Block.

**Comment by seanvandermolen** [comment]
Thank you, that makes sense.
